recommendation_engine/data_store/abstract_data_store.py

```diff
@@ -0,0 +1,38 @@
+import abc
+
+
+class AbstractDataStore(object):
```

Comment:
I think you need to inherit `abc.ABC` class also to make this class an abstract class and no need to inherit `object` class unless you also want to make your code python 2 compatible also.